cds-snc/notification-api
# Auto assign the notify organisation when a service is first created

# Summary | Résumé

Auto assign the notify organisation when a service is first created.
This required some changes to the gc-organisations repo to add the Notify organisation ID to the dataset: https://github.com/cds-snc/gc-organisations/pull/47

I also added some logic to set the 3 day data retention if the service belongs to a Prov/Territory.

# Testing

To test this locally you will need to do some set up.

## Set up:
1. check the branch out locally
2. Add the 2 new environment variables to your .env file: `CRM_GITHUB_PERSONAL_ACCESS_TOKEN` and `CRM_ORG_LIST_URL`. You can find the values in the notification-manifest repo
3. run the api and admin locally
4. log in as a platform admin and create an organisation called "Government of Ontario - Local Test"
5. open your local db via the Postico App and find the new record in the `organisation` table
6. edit and save a new `id` value for your new organisation: `c4be8612-85c3-454b-ace1-309352220cf5`. This is so it matches up the organisation ID in prod
7. visit the page http://localhost:6012/organisations/c4be8612-85c3-454b-ace1-309352220cf5 to verify that your new id has been applied correctly

## Test 1:
1. Create a new service and select "Government of Ontario" from the dropdown. 
2. Enter "test" in the box for "Enter any other names for your group (Optional)"
3. Create the service
4. go to the service settings page and verify that the organisation has been set correctly as "Government of Ontario - Local Test"

## Test 2:
Same steps as Test 1, but go through the service creation flow in french

## Test 3:
Same steps as Test 1 but select "Government of British Columbia". No organisation should be set automatically since there is no local organisation with the same ID.
